src/kOS/Module/kOSCustomParameters.cs

```diff
@@ -0,0 +1,187 @@
+﻿using KSP.IO;
+using System.Reflection;
+
+namespace kOS.Module
+{
+    public class kOSCustomParameters : GameParameters.CustomParameterNode
+    {
+        private static kOSCustomParameters instance;
+
+        public static kOSCustomParameters Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    instance = HighLogic.CurrentGame.Parameters.CustomParams<kOSCustomParameters>();
+                }
+                return instance;
+            }
+        }
+
+        public const string MIGRATION_DIALOG_TEXT = "Some kOS settings are now stored in the KSP save file instead " +
+            "of an external file. Telnet settings are still stored in the external file. It appears that you still " +
+            "have valid settings stored in the external file, but that this is either a new save or the save has " +
+            "not yet been migrated.  Would you like to migrate now?";
+
+        [GameParameters.CustomParameterUI("")]
+        public bool migrated = false;
+
+        [GameParameters.CustomIntParameterUI("")]
+        public int version = 0;
+
+        [GameParameters.CustomIntParameterUI("Instructions per update", minValue = 50, maxValue = 2000)]
+        public int instructionsPerUpdate = 200;
+
+        [GameParameters.CustomParameterUI("Enable compressed storage")]
+        public bool useCompressedPersistence = true;
+
+        [GameParameters.CustomParameterUI("Show statistics")]
+        public bool showStatistics = false;
+
+        [GameParameters.CustomParameterUI("Enable Remote Tech integration")]
+        public bool enableRTIntegration = true;
+
+        [GameParameters.CustomParameterUI("Start on the archive")]
+        public bool startOnArchive = false;
+
+        [GameParameters.CustomParameterUI("Obey hide UI toggle")]
+        public bool obeyHideUi = true;
+
+        [GameParameters.CustomParameterUI("Enable safe mode")]
+        public bool enableSafeMode = true;
```

Comment:
We can definitely do that with tooltips.  I think that it's another parameter on the attribute.